web/src/settings_notifications.js

```diff
@@ -121,18 +121,20 @@ export function set_enable_marketing_emails_visibility() {
 export function set_up(settings_panel) {
     const $container = $(settings_panel.container);
     const settings_object = settings_panel.settings_object;
-    const $notification_sound_elem = $(settings_panel.notification_sound_elem);
+    const $temp_notification_sound_elem = $("#temp-realm-default-notification-sound-audio");
     const for_realm_settings = settings_panel.for_realm_settings;
+    const $notification_sound_dropdown = $container.find(".setting_notification_sound");
 
     $container.find(".play_notification_sound").on("click", () => {
-        if (settings_object.notification_sound !== "none") {
-            ui_util.play_audio($notification_sound_elem[0]);
+        const sound = $notification_sound_dropdown.val().toLowerCase();
+        if (settings_object.notification_sound !== "none" && sound !== "none") {
+            notifications.update_notification_sound_source($temp_notification_sound_elem, { notification_sound: sound });
```

Comment:
Would it instead be better to just call `update_notification_sound_source` when changing the dropdown val using `.on("change")` handler and then reset it again when a user click on "Discard" button? I think this would be much simpler to understand for anyone who reads the code and seems a better approach.

We would not need to define an extra `<audio>` element as well. We can use the existing `#realm-default-notification-sound-audio` element as it is only used to play the sound and nothing else.

Reply:
I have fixed it using ```<audio id="realm-default-notification-sound-audio">``` instead of extra ```<audio>``` tag such that whenever play-button is clicked, ```<audio id="realm-default-notification-sound-audio">``` is changed according to the dropdown value selected.